### PR TITLE
change(authz-keycloak): remove deprecated audience

### PR DESCRIPTION
**Pull request by tzssangglass**
### Description

Fixes #7482

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: t/plugin/authz-keycloak.t [hunk 1]
```diff
@@ -74,32 +74,7 @@ done
 
 
 
-=== TEST 3: minimal valid configuration with audience
---- config
-    location /t {
-        content_by_lua_block {
-            local plugin = require("apisix.plugins.authz-keycloak")
-            local ok, err = plugin.check_schema({
-                                audience = "foo",
```
Comment (spacewander) [inline t/plugin/authz-keycloak.t:83]: Better to change it to use client_id instead of removing the test?

Comment (tzssangglass) [reply t/plugin/authz-keycloak.t:83]: we have a test case cover `client_id`, ref: https://github.com/apache/apisix/blob/d3d6151c80fcdc162401429bb0920b381d3f2dd7/t/plugin/authz-keycloak.t#L52-L73

so just remove this is ok.

File: docs/en/latest/plugins/authz-keycloak.md [hunk 2]
```diff
@@ -46,8 +46,7 @@ Refer to [Authorization Services Guide](https://www.keycloak.org/docs/latest/aut
...
-| audience                                     | string        | False    |                                               |                                                                    | Legacy parameter now replaced by `client_id` kept for backwards compatibility. Either `client_id` or `audience` is required.                                                                                                                          |
+| client_id                                    | string        | False    |                                               |                                                                    | The identifier of the resource server to which the client is seeking access.                                                                                                                                                                         |
```
Comment (tokers) [inline docs/en/latest/plugins/authz-keycloak.md:49]: Now `client_id` is required.

Comment (tzssangglass) [reply docs/en/latest/plugins/authz-keycloak.md]: fixed

**Review by spacewander — APPROVED** [review]

File: docs/zh/latest/plugins/authz-keycloak.md [hunk 3]
```diff
@@ -73,7 +72,7 @@ description: 本文介绍了关于 Apache APISIX `authz-keycloak` 插件的基
     - 使用 `discovery` 属性后，`authz-keycloak` 插件就可以从其 URL 中发现 Keycloak API 的端点。该 URL 指向 Keyloak 针对相应领域授权服务的发现文档。
     - 如果发现文档可用，则插件将根据该文档确定令牌端点 URL。如果 URL 存在，则 `token_endpoint` 和 `resource_registration_endpoint` 的值将被其覆盖。
 - Client ID and secret
-    - 该插件需配置 `client_id` 或 `audience`（用于向后兼容）属性来标识自身，如果两者都已经配置，则 `client_id` 优先级更高。
+    - 该插件需配置 `client_id` 属性来标识自身，如果两者都已经配置，则 `client_id` 优先级更高。
```
Comment (starsz) [inline docs/zh/latest/plugins/authz-keycloak.md:75]:
```suggestion
    - 该插件需配置 `client_id` 属性来标识自身
```

Comment (tzssangglass) [reply docs/zh/latest/plugins/authz-keycloak.md]: done

**Review by spacewander — APPROVED** [review]

**Review by starsz — APPROVED** [review]

**Review by tokers — APPROVED** [review]

**Pull request merged**